src/containers/StructurePage/folderComponents/menuOptions/ToggleVisibility.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2017-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import { injectT } from '@ndla/i18n';
+import { Eye } from '@ndla/icons/editor';
+import styled from '@emotion/styled';
+import { spacing } from '@ndla/core';
+
+import {
+  updateSubjectMetadata,
+  updateTopicMetadata,
+} from '../../../../modules/taxonomy';
+import RoundIcon from '../../../../components/RoundIcon';
+import ToggleSwitch from '../../../../components/ToggleSwitch';
+import { TranslateType } from '../../../../interfaces';
+import MenuItemButton from './MenuItemButton';
+
+enum MenuType {
+  subject = 'subject',
+  topic = 'topic',
+}
+
+interface Props {
+  id: string;
+  editMode: string;
+  name: string;
+  metadata: { grepCodes: string[]; visible: boolean };
+  t: TranslateType;
+  toggleEditMode: Function;
+  menuType: MenuType;
+}
+
+export const DropDownWrapper = styled('div')`
+  display: flex;
+  justify-content: space-between;
+  font-size: 0.9rem;
+  background-color: white;
+  padding: calc(${spacing.small} / 2);
+`;
+
+const ToggleVisibility: FC<Props> = ({
+  id,
+  editMode,
+  name,
+  metadata,
+  t,
+  toggleEditMode,
+  menuType,
+}) => {
+  const [visible, setVisible] = useState(metadata?.visible);
+
+  const updateMetadata = async (visible: boolean) => {
+    switch (menuType) {
+      case 'subject': {
+        await updateSubjectMetadata(id, {
+          grepCodes: metadata.grepCodes,
+          visible: !visible,
+        });
+        setVisible(!visible);
+        break;
+      }
+
+      case 'topic': {
+        await updateTopicMetadata(id, {
+          grepCodes: metadata.grepCodes,
+          visible: !visible,
+        });
+        setVisible(!visible);
+        break;
+      }
+
+      default:
+        return null;
+    }
+  };
+
+  const toggleEditModes = () => {
+    toggleEditMode('toggleMetadataVisibility');
+  };
+
+  useEffect(() => {}, [editMode]);
+
+  const ToggleMenu = visible ? (
+    <DropDownWrapper>
+      {name} {t('metadata.visible')}
+      <ToggleSwitch
```

Comment:
Denne har predefinert valga T og K som står for tilleggstoff og kjernestoff. Ser at disse er definert med content: 'T'; og content: 'K'; Du bør endre slik at disse verdiene kan sendes inn som props for at du kan overstyre dei.